library/systemd/test/systemd_service_test.rb

```diff
@@ -130,6 +130,201 @@ module Yast
+        end
+      end
+
+      context "when the service is disabled" do
+        let(:enabled_on_boot?) { false }
+
+        context "but the associated socket is enabled" do
+          it "returns :demand" do
+            expect(service.start_mode).to eq(:demand)
+          end
+        end
+
+        context "and the socket is disabled" do
+          let(:socket) { double("socket", enabled?: false) }
+
+          it "returns :manual" do
+            expect(service.start_mode).to eq(:manual)
+          end
+        end
+
+        context "and there is no socket" do
+          let(:socket) { nil }
+
+          it "returns :manual" do
+            expect(service.start_mode).to eq(:manual)
+          end
+        end
+      end
+    end
+
+    describe "#start_mode=" do
+      subject(:service) { SystemdService.find("cups") }
+      let(:socket) { double("socket", disable: true) }
+
+      before do
+        stub_services(service: "cups")
+        allow(service).to receive(:socket).and_return(socket)
+        allow(service).to receive(:disable)
+      end
+
+      context "when no argument is given" do
+        it "enables the service to start on boot" do
+          expect(socket).to_not receive(:enable)
+          service.start_mode = :boot
+        end
+      end
+
+      context "when :boot mode is given" do
+        it "enables the service to start on boot" do
+          expect(service).to receive(:enable_service)
+          expect(socket).to_not receive(:enable)
+          service.start_mode = :boot
+        end
+      end
+
+      context "when :demand mode is given" do
+        it "enables the socket" do
+          expect(service).to_not receive(:enable_service)
+          expect(socket).to receive(:enable)
+          service.start_mode = :demand
+        end
+      end
+
+      context "when :manual mode is given" do
+        it "disables the service and the socket" do
+          expect(service).to receive(:disable_service)
+          expect(socket).to receive(:disable)
+          service.start_mode = :manual
+        end
+      end
+    end
+
+    describe "#start_modes" do
+      subject(:service) { SystemdService.find("cups") }
+
+      before do
+        stub_services(service: "cups")
+        allow(service).to receive(:socket).and_return(socket)
+      end
+
+      context "when an associated socket exists" do
+        let(:socket) { double("socket", disable: true) }
+
+        it "returns :boot, :demand and :manual" do
+          expect(service.start_modes).to eq([:boot, :demand, :manual])
+        end
+      end
+
+      context "when no associated socket exists" do
+        let(:socket) { nil }
+
+        it "returns :boot and :manual" do
+          expect(service.start_modes).to eq([:boot, :manual])
+        end
+      end
+    end
+
+    describe "#enable" do
+      subject(:service) { SystemdService.find("cups") }
+
+      before do
+        stub_services(service: "cups")
+      end
+
+      it "sets start_mode to :boot" do
```

Comment:
I am not sure if this is good test and documention. You want to be sure that when you call enable that it "enable service". This is important and not some internal method for start_mode.

Reply:
I wanted to avoid repetition. To be honest, `enable` now sets the start mode to `:boot` (that's what it does).

Reply:
well, but we have this method just to be backward compatible. So we should be sure it do what it did before. So we do not break other modules.